### PR TITLE
Remove internal usage of Logger in extension projects and remove Initialize(Logger) from IExternalSerializer

### PR DESCRIPTION
**Pull request by xiazen**
- Remove internal usage of Logger in extension projects
- Remove `Initialize(Logger logger)` method from `IExternalSerializer` interface

### CODE REVIEW

File: src/Orleans.Core/Serialization/IExternalSerializer.cs [hunk 1]
```diff
@@ -13,12 +12,6 @@ namespace Orleans.Serialization
     public interface IExternalSerializer
```
Comment (ReubenBond) [inline src/Orleans.Core/Serialization/IExternalSerializer.cs]: This is a breaking change, we need to make sure that it's called out in the release notes.

Comment (xiazen) [reply src/Orleans.Core/Serialization/IExternalSerializer.cs]: @jdom to notify

File: src/OrleansSQLUtils/Storage/Provider/AdoNetStorageProvider.cs [hunk 2]
```diff
@@ -200,10 +202,12 @@ public class AdoNetStorageProvider: IStorageProvider
                 queries.Single(i => i.Item1 == "ReadFromStorageKey").Item2,
                 queries.Single(i => i.Item1 == "ClearStorageKey").Item2);
 
-            Log = providerRuntime.GetLogger(GetType().FullName);
+            var loggerFactory = providerRuntime.ServiceProvider.GetService<ILoggerFactory>();
+            this.logger = loggerFactory.CreateLogger<AdoNetStorageProvider>();
+            Log = new LoggerWrapper(this.logger, this.GetType().FullName, loggerFactory);
```
Comment (ReubenBond) [inline src/OrleansSQLUtils/Storage/Provider/AdoNetStorageProvider.cs]: Do we need the wrapper here?

You should be able to get `ILogger<AdoNetStorageProvider>` from the container directly, with no factory.

Comment (xiazen) [reply src/OrleansSQLUtils/Storage/Provider/AdoNetStorageProvider.cs]: that is for IStorageProvider.Logger method. There's discussion about moving that to a legacy package, or even remove it. But I don't want to involve that in this PR yet.

Comment (xiazen) [reply src/OrleansSQLUtils/Storage/Provider/AdoNetStorageProvider.cs]: This is the same with what I did in #3611

**Pull request merged**